frontend/test/lit/test_mlir_decomposition.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file performs the frontend lit tests that the peephole transformations are correctly lowered.
+
+We check the transform jax primitives for each pass is correctly injected
+during tracing, and these transform primitives are correctly lowered to the mlir before
+running -apply-transform-sequence. 
+"""
+
+# RUN: %PYTHON %s | FileCheck %s
+
+# pylint: disable=line-too-long
+
+import platform
+import shutil
+
+import pennylane as qml
+from lit_util_printers import print_jaxpr
+
+from catalyst import qjit
+from catalyst.debug import get_compilation_stage
+from catalyst.device import get_device_capabilities
+from catalyst.utils.runtime_environment import get_lib_path
+
+
+def flush_peephole_opted_mlir_to_iostream(QJIT):
```

Comment:
`QJIT` is just a local variable right? I think we should rename it to avoid confusion with the `QJIT` class.

Reply:
Yes I copied it from another test, I have renamed it